compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEAttrs.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===- IREECodegenAttrs.h - Codegen dialect attributes --------------------===//
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_COMPILER_AMDAIE_DIALECT_PACKINGCONFIG_H_
+#define IREE_COMPILER_AMDAIE_DIALECT_PACKINGCONFIG_H_
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace mlir::iree_compiler {
+struct PackConfigTD {
+  // Expected packed sizes for specified iterator dimensions
+  ArrayRef<int64_t> packedSizes;
+  // Indices of pack operations need to be transposed
+  ArrayRef<int64_t> transposePackIndices;
+  // Indicator of if there is a unpack op corresponding to a pack op
+  ArrayRef<int64_t> unpackEmpty;
+  // Attributes for inner dimension permutation
+  ArrayRef<ArrayRef<int64_t>> innerPerm;
+  // Attributes for outer dimension permutation
+  ArrayRef<ArrayRef<int64_t>> outerPerm;
+};
+/// Typedef for packing config to use at different levels of packing.
+using PackingConfigListType = SmallVector<PackConfigTD>;
+using PackingConfigListTypeRef = ArrayRef<PackConfigTD>;
+struct PackConfig {
```

Comment:
PackConfig vs PackConfigTD

This is a stopgap solution.

I wanted to make `packedSizes` as an Array of `I64Attr` -> but for some reason `I64Attr` remained unknown (I tried plausible `.h` and `.td` inclusions but to no avail).

I then tried `IntegerAttr` instead of `I64Attr` in the `.td` - it didn't lead to the previous error but the following became an issue :-

OpFoldResult is NOT Attribute.

I felt spent trying to fight this (yesterday and today) and my first intention was to at least make it work.
Hence I added this as a stopgap solution.

Let me know if you have any pointers regarding the same.
One of the major issues that I faced was the nuances involving : ArrayRef <-> SmallVector (and with the multi-dimensional vectors/arrays it piled up the issues).

Reply:
I see your reason for setting PackConfig vs PackConfigTD, however we should try to remove one of these. I can help take a look at the issue.

Reply:
I added a commit using one PackConfig struct.

Reply:
I think we dont need this struct here. The same information is carried by the `Attribute`. The `Attribute` is also a C++ struct. Its definition should have accessors you need to get the information. We dont need to unpack the attribute into this struct.